cmd/proxy/Dockerfile

```diff
@@ -1,10 +1,11 @@
-FROM golang:1.11
+FROM golang:1.11-alpine
 
 RUN mkdir -p $GOPATH/src/github.com/gomods/athens
 WORKDIR $GOPATH/src/github.com/gomods/athens
 
 COPY . .
 
+ENV GO111MODULE=on
```

Comment:
you can just have this be in the next line: 
`RUN GO111MODULE=on CGO_ENABLED=0 go build -mod=vendor -o /bin/app ./cmd/proxy` 

No need to hash a whole step for it.